### PR TITLE
added ORDER BY RAND() to add complete shuffling when creating dataset with BQ

### PR DESCRIPTION
**Pull request by takumiohym**
Fix related to issue #1376.

Not sure this solution is following best practices, but I think this is the easiest workaround.

I modified data creation part under `building_production_ml_systems` and `structured` directory, but please let me know I should fix other notebooks too.

### CODE REVIEW

**Comment by danielsparing** [comment]
In my understanding, for a relational database incl. BigQuery, we should not specify an `ORDER BY` in a `CREATE TABLE` statement, as the order of rows is not guaranteed at all when reading the table.

Why do you need the CSV's to be shuffled in the first place though? At first glance, when we later read them in using [`make_csv_dataset`](https://www.tensorflow.org/api_docs/python/tf/data/experimental/make_csv_dataset), this should by default shuffle the rows.

**Comment by takumiohym** [comment]
Hi Daniel,
Thank you for your comment.

This issue is also written here.
We can apply shuffling in Tensorflow dataset API, but since this is not a complete shuffling but partial shuffling in specified buffer size, this still cause overfitting problem.

Actually this lab is facing overfitting by default.

WIthout shuffling: 
Screen Shot 2021-02-02 at 18 56 25
https://tensorboard.dev/experiment/LpoN6pExS762gbILkfaMgA/#scalars&_smoothingWeight=0.206

With shuffling by `ORDER BY RAND()`:
Screen Shot 2021-02-02 at 19 27 54
https://tensorboard.dev/experiment/EwGVu90gQE2Lokv90rEmXg/#scalars

What do you think?

**Comment by danielsparing** [comment]
I had a second look at this issue and I am not able to reproduce the overfitting above (even though I was able to reproduce the matplotlib histograms at https://github.com/GoogleCloudPlatform/training-data-analyst/issues/1376 ).

That said, I still think that adding the RAND() to the _table definition_ is not the way to go, and given that the goal of the lab is to illustrate the functionality of submitting a run to the cloud, might not even be necessary.

Things we _could_ do:
- adding `tf.data.experimental.make_csv_dataset( ... , shuffle_buffer_size=1000000)` and/or increasing to `NUM_EXAMPLES_TO_TRAIN_ON=5000` or higher improves things, with no substantial slowdown (but then again I couldn't reproduce the overfit)
- as I mentioned earlier we could rewrite the `bq extract` into a `bq query` reading a `view` including the `ORDER BY RAND()`, but this slightly adds to the complexity of the lab, with probably minimal slowdown again.
